src/background.js

```diff
@@ -196,3 +199,20 @@ browser.storage.local.get()
 browser.browserAction.onClicked.addListener(() => {
     browser.sidebarAction.open();
 });
+
+// context menu for 'Send to Notes'
+browser.contextMenus.create({
+  id: 'send-to-notes',
+  title: browser.i18n.getMessage('sendToNotes'),
+  contexts: ['selection']
+});
+
+browser.contextMenus.onClicked.addListener((info) => {
+  browser.sidebarAction.open();
+  setTimeout(() => {
+    chrome.runtime.sendMessage({
+      action: 'send-to-notes',
+      text: info.selectionText
+    });
+  }, 700);
```

Comment:
what is the this `700` magic number? would this cause issues on slower machines? Might be worth commenting this code for the future :)

Reply:
@cedricium can you do:

```js
browser.sidebarAction.open().then(() => {
   chrome.runtime.sendMessage({....})
   // rest of the code 
})
```

Reply:
Doesn't work - the Promise resolves quicker than the actual sidebar opening. Somehow I need to:

1. call `browser.sidebarAction.open()`
2. verify sidebar is open **AND** the ckeditor instance is ready
3. wait for the above step before...
4. sending message to `panel.js` to paste selected text

Let me think about this, I'll see if I can figure it out.

Reply:
When CKEditor is ready it fires an event to background (i think it already does that). maybe it can fetch back the text from the `selectionText`

Reply:
Do you know where that event is registered in or listened for in either `background.js` or `panel.js`?

Reply:
https://github.com/mozilla/notes/blob/master/src/sidebar/panel.js#L218

Reply:
there is a `kinto-load` via loadContent that gets called when editor is ready i think